staging/src/k8s.io/api/autoscaling/v2beta1/types.go

```diff
@@ -178,6 +204,39 @@ type ResourceMetricSource struct {
 	TargetAverageValue *resource.Quantity `json:"targetAverageValue,omitempty" protobuf:"bytes,3,opt,name=targetAverageValue"`
 }
 
+// PodResourceMetricSource indicates how to scale on a resource metric known to
+// Kubernetes, as specified in requests and limits, describing each pod in the
+// current scale target (e.g. CPU or memory).  The values will be averaged
+// together before being compared to the target.  Such metrics are built in to
+// Kubernetes, and have special scaling options on top of those available to
+// normal per-pod metrics using the "pods" source.  Only one "target" type
+// should be set.
+type PodResourceMetricSource ResourceMetricSource
+
+// ContainerResourceMetricSource indicates how to scale on a resource metric known to
+// Kubernetes, as specified in requests and limits, describing each pod in the
+// current scale target (e.g. CPU or memory).  The values will be averaged
+// together before being compared to the target.  Such metrics are built in to
+// Kubernetes, and have special scaling options on top of those available to
+// normal per-pod metrics using the "pods" source.  Only one "target" type
+// should be set.
+type ContainerResourceMetricSource struct {
```

Comment:
is there a reason not to use the same schema as v2beta2 ContainerResourceMetricSource here? that would simplify (and maybe eliminate the need for) custom conversion methods. same question applies to v1 ContainerResourceMetricSource and the v1/v2beta1 ContainerResourceMetricStatus types

Reply:
It looks like the schema in v2beta2 uses MetricTarget which only exists in v2beta2. Would it be worth copying that to v1/v2beta1 for use here? I don't have a strong opinion whether it's better to keep the v1/v2beta1 bit consistent with existing v1/v2beta1 metric specs, or to keep this new field consistent across versions.

Reply:
v2beta2 metric target was expanded and unified averageUtilization/averageValue... that doesn't exist in v1 or v2beta1